doc/congsemigraph.xml

```diff
@@ -0,0 +1,146 @@
+############################################################################
+##
+##  congsemigraph.xml
+##  Copyright (C) 2021                       Marina Anagnostopoulou-Merkouri
+##                                                            James Mitchell 
+##
+##  Licensing information can be found in the README file of this package.
+##
+############################################################################
+
+<#GAPDoc Label="IsCongruenceByWangPair">
+  <ManSection>
+    <Prop Name="IsCongruenceByWangPair" Arg="cong"/>
+    <Description>
+    A congruence by Wang pair <C>cong</C> is a congruence of a graph inverse semigroup <C>S</C>
+    which is expressed in terms of two sets <A>H</A> and <A>W</A> of vertices of the corresponding
+    graph of <A>S</A>. For more information on Wang pairs see <Cite Key="Wang2019aa"/> and
+    <Cite Key="Anagnostopoulou-Merkouri2021aa"/>.<P/>
+
+    <Example><![CDATA[
+gap> D := Digraph([[3, 4], [3, 4], [4], []]);
+<immutable digraph with 4 vertices, 5 edges>
+gap> S := GraphInverseSemigroup(D);
+<finite graph inverse semigroup with 4 vertices, 5 edges>
+gap> cong := CongruenceByWangPair(S, [3, 4], []);
+<gis-congruence with H=[ 3, 4 ], W=[  ]>
+gap> IsCongruenceByWangPair(cong);
+true
+gap> cong := CongruenceByWangPair(S, [4], [2]);
+<gis-congruence with H=[ 4 ], W=[ 2 ]>
+gap> IsCongruenceByWangPair(cong);
+true
+gap> e_1 := S.1;
+e_1
+gap> e_3 := S.3;
+e_3
+gap> cong := SemigroupCongruence(S, [[e_1, e_3]]);
+<semigroup congruence over <finite graph inverse semigroup with 4 
+ vertices, 5 edges> with 1 generating pairs>
+gap> IsCongruenceByWangPair(cong);
+false
+]]></Example>
+    </Description>
+  </ManSection>
+  <#/GAPDoc>
+
+<#GAPDoc Label="CongruenceByWangPair">
+  <ManSection>
+    <Func Name = "CongruenceByWangPair" Arg = "S, H, W"/>
+    <Returns>A semigroup congruence.</Returns>
+    <Description>
+      This function returns a semigroup congruence over the graph inverse semigroup
+      <A>S</A> in the form of a Wang pair.<P/>
```

Comment:
Isn't it necessary for `S` to be finite for any of this to work? If so, this should be documented and checked in the code.